pkg/downloader/manager.go

```diff
@@ -191,13 +191,13 @@ func (m *Manager) Update() error {
 
 	// Now we need to find out which version of a chart best satisfies the
 	// dependencies in the Chart.yaml
-	lock, err := m.resolve(req, repoNames)
+	lock, urls, err := m.resolve(req, repoNames)
 	if err != nil {
 		return err
 	}
```

Comment:
Do we need to check that `urls` is empty?

Reply:
`urls` is ultimately read on line 756, if it is empty the key will not be found and the flow will proceed to the else statement and proceed as per existing implementation without any issues, so I am not seeing a reason to check for empty